### ISSUE TITLE
URL object as a prop

### ISSUE THREAD

**Issue by martpie** [isaachinman/next-i18next#517]
## Is your feature request related to a problem? Please describe.
The problem is related to the error `'as' type must be 'string', but it is object`

## Describe the solution you'd like
Next.js's Link can accept more than a string for the `as` property, it can be a `UrlLike`

For example: `<Link href={...} as={{ pathname: '/test', query: { page: 1 } }}` with output `<a href="/test?page=1">`.

It would be neat to have it in `next-i18next` as well

## Describe alternatives you've considered

The usual `<Link href={...} as={'/test?page=1'}`, but it's a bit more complex to manipulate with dynamic data.

## Additional context
n/a

**Comment by isaachinman** [comment]
Sure, we'd need to somehow handle that case [here](https://github.com/isaachinman/next-i18next/blob/09be96fad33d37f38997cf89103a4b7d690518a8/src/utils/lng-path-corrector.ts#L17).

I don't personally require this use case - open to contributions.

**Comment by martpie** [comment]
should not be too hard, I can have a look at it myself and will submit a PR :)

**Comment by martpie** [comment]
The new documentation makes it more explicit how to use URL objects: https://nextjs.org/docs/api-reference/next/router#with-url-object

I'll try to work on this after #581 is merged.

**Comment by isaachinman** [comment]
That would be great, thanks!

**Comment by Kenpa** [comment]
Hey, any update on this feature?

**Comment by isaachinman** [comment]
Hi there, I have just released `next-i18next@8.0.0`, which uses the internationalised routing built into NextJs v10+. This new version of `next-i18next` is quite different, thus making this PR no longer relevant. Thanks!
